### PR TITLE
Add support for Bazel build targets and fix some tests.

### PR DESCRIPTION
**Pull request by jraisher**
Not all unit tests are included in the new build file since the
tests are not all simple C++ tests.

### CODE REVIEW

**Comment by jraisher** [comment]
Thanks for the comments, I've incorporated them and updated the CR.

File: BUILD [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+    ]),
+    includes = [
+        "googletest",
+        "googletest/include",
+    ],
+    copts = [
+        "-g",
+        "-Wall",
+        "-Wextra",
+<<<<<<< HEAD
```
Comment (balasanjay) [inline BUILD]: Bad merge, here and below.

File: BUILD [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+licenses(["notice"])
+
+cc_library(
+    name = "googletest",
+    srcs = glob([
+    	"googletest/src/gtest.cc",
+    	"googletest/src/gtest-death-test.cc",
+    	"googletest/src/gtest-filepath.cc",
+    	"googletest/src/gtest-internal-inl.h",
+    	"googletest/src/gtest_main.cc",
```
Comment (balasanjay) [inline BUILD]: Can you split gtest_main.cc into its own separate target?

We have a library that needs to include gtest to get the FRIEND_TEST macro, but we don't want a main in that situation.

**Comment by gennadiycivil** [comment]
I see failed Travis build https://travis-ci.org/google/googletest/builds/80834800.
Please review if this is still relevant.

**Pull request closed without merging**